backend/src/controllers/MyRestaurantController.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from 'express';
+import Restaurant from '../models/restaurant';
+import mongoose from 'mongoose';
+import { v2 as cloudinary } from 'cloudinary';
+
+const createMyRestaurant = async (req: Request, res: Response) => {
+  try {
+    const existingRestaurant = await Restaurant.findOne({ user: req.userId });
+    if (existingRestaurant) {
+      return res.status(409).json({ message: 'Restaurant already exists' });
+    }
+
+    const image = req.file as Express.Multer.File;
+    const base64Image = image.buffer.toString('base64');
+    const dataURI = `data:${image.mimetype};base64,${base64Image}`;
+    const uploadResponse = await cloudinary.uploader.upload(dataURI);
+
+    const restaurant = new Restaurant(req.body);
+    restaurant.imageFile = uploadResponse.secure_url;
+    restaurant.user = new mongoose.Types.ObjectId(req.userId);
+
+    await restaurant.save();
+    res.status(201).send(restaurant);
+  } catch (error) {
+    console.log(error);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/27#issuecomment-2383391092>#6fe9f8</a><span> - 09/30/2024, 02:40 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Inadequate error logging in the catch block
</div>
<div id="fix">
<b>Fix:</b> Implement more detailed error logging, including the error message and stack trace. Consider using a logging library like Winston for more advanced logging capabilities.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -22,8 +22,8 @@
    await restaurant.save();
    res.status(201).send(restaurant);
  } catch (error) {
 -  console.log(error);
 -  res.status(500).json({ message: 'Error creating restaurant' });
 +  console.error('Error creating restaurant:', error);
 +  res.status(500).json({ message: 'An unexpected error occurred while creating the restaurant' });
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged